ambry-router/src/test/java/com.github.ambry.router/MockServer.java

```diff
@@ -55,8 +58,11 @@
   private ServerErrorCode hardError = null;
   private LinkedList<ServerErrorCode> serverErrors = new LinkedList<ServerErrorCode>();
   private final Map<String, ByteBuffer> blobs = new ConcurrentHashMap<String, ByteBuffer>();
-  private final HashMap<String, ServerErrorCode> blobIdToServerErrorCode = new HashMap<String, ServerErrorCode>();
+  private final Set<String> dataBlobs = Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
```

Comment:
see if it is better to simply store the `BlobType` along with the `ByteBuffer` in a simple class and associate that with the blob id in the `blobs` HashMap. That way you could avoid `dataBlobs`.